src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+
+import static java.lang.Integer.parseInt;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DateCommand;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+
+public class Parser {
+
+    /**
+     * Parses the command entered by the user and returns a executable
+     * command object.
+     *
+     * @param command The command entered by the user.
+     * @return A command object that can be executed.
+     * @throws DukeException If the command is invalid
+     */
+    public static Command parse(String command) throws DukeException {
```

Comment:
This method seems a bit long as stated by the [code quality](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods).
Maybe the codes under different cases (eg. `command.startsWith("deadline")`, `command.startsWith("event")` etc.) can each be extracted out into a method?

Reply:
I really like the idea of abstracting out the different cases into their own separate methods. Thank you for your suggestion!